reader.go

```diff
@@ -0,0 +1,121 @@
+package recordio
+
+import (
+	"io"
+	"log"
+)
+
+// Index consists offsets and sizes of the consequetive chunks in a RecordIO file.
+type Index struct {
+	chunkOffsets []int64
+	chunkLens    []uint32
+	numRecords   int // the number of all records in a file.
+}
+
+// LoadIndex scans the file and parse chunkOffsets, chunkLens, and len.
+func LoadIndex(r io.ReadSeeker) (*Index, error) {
+	f := &Index{}
+	offset := int64(0)
+	var e error
+	var hdr *Header
+
+	for {
+		if hdr, e = parseHeader(r); e != nil {
+			break
+		} else {
```

Comment:
golang style prefer early return than if {} else {}.
I think following is more clear (have less indentation):

```go
for {
  if {
    break
  }
  // no else, more code here.
}
```